resource/environment/integration.js

```diff
@@ -54,38 +19,73 @@ function onFailure( err ) {
 	};
 }
 
-function onSuccess( data ) {
-	return { data: data };
-}
+const connectionError = {
+	status: 500,
+	data: {
+		message: "Could not connect to Rancher instance."
+	}
+};
 
-function onUpdated( env ) {
-	return {
-		status: 200,
-		data: env
-	};
-}
+const readError = {
+	status: 404,
+	data: {
+		message: "Unable to get information from the database"
+	}
+};
+
+const dockerError = {
+	data: {
+		message: "Validation with Dockerhub failed."
+	},
+	status: 401
+};
+
+const tagNotFoundError = {
+	data: {
+		message: "Image does not exist in Dockerhub"
+	},
+	status: 404
+};
 
-function onCreated() {
+function invaildTagError(image) {
```

Comment:
@EvanSnapp - minor typo here - need to change this (and references to it) to `invalidTagError` :)